website/project/views/node.py

```diff
@@ -475,6 +475,9 @@ def project_set_privacy(auth, node, **kwargs):
     if permissions is None:
         raise HTTPError(http.BAD_REQUEST)
 
+    if permissions == 'private' and node.is_registration and not node.registered_before_cutoff_date:
+        raise HTTPError(http.BAD_REQUEST)
```

Comment:
Please add a useful message by passing `data=dict(message_long='Public registrations cannot be made private.')`